apps/etterlatte-vedtaksvurdering/src/main/kotlin/vedtaksvurdering/database/VedtaksvurderingRepository.kt

```diff
@@ -39,234 +36,150 @@ class VedtaksvurderingRepository(private val datasource: DataSource) {
+        params = mapOf(
+            "saksbehandlerId" to saksbehandlerId,
+            "vedtakfattet" to true,
+            "vedtakstatus" to VedtakStatus.FATTET_VEDTAK.name,
+            "behandlingId" to behandlingsId
+        ),
+        loggtekst = "Fatter vedtok for behandling $behandlingsId"
+    )
 
     fun attesterVedtak(
         saksbehandlerId: String,
         behandlingsId: UUID,
         vedtakId: Long,
         utbetalingsperioder: List<Utbetalingsperiode>
     ) {
-        connection.use {
-            val insertPersoderStatement = it.prepareStatement(Queries.lagreUtbetalingsperiode)
-            utbetalingsperioder.forEach {
-                insertPersoderStatement.setLong(1, vedtakId)
-                insertPersoderStatement.setDate(2, it.periode.fom.atDay(1).let(Date::valueOf))
-                insertPersoderStatement.setDate(3, it.periode.tom?.atEndOfMonth()?.let(Date::valueOf))
-                insertPersoderStatement.setString(4, it.type.name)
-                insertPersoderStatement.setBigDecimal(5, it.beloep)
-
-                insertPersoderStatement.addBatch()
-            }
-            if (utbetalingsperioder.isNotEmpty()) {
-                insertPersoderStatement.executeBatch().forEach {
-                    require(it == 1)
-                }
-            }
-
-            val statement = it.prepareStatement(Queries.attesterVedtak)
-            statement.setString(1, saksbehandlerId)
-            statement.setString(2, VedtakStatus.ATTESTERT.name)
-            statement.setObject(3, behandlingsId)
-            statement.execute()
-        }
-    }
-
-    fun underkjennVedtak(
-        behandlingsId: UUID
-    ) {
-        connection.use {
-            val statement = it.prepareStatement(Queries.underkjennVedtak)
-            statement.setString(1, VedtakStatus.RETURNERT.name)
-            statement.setObject(2, behandlingsId)
-            statement.execute()
-        }
-    }
-
-    fun slettSak(sakId: Long) {
-        connection.use {
-            it.prepareStatement(Queries.slettUtbetalingsperioderISak).apply {
-                setLong(1, sakId)
-                executeUpdate()
-            }
-            it.prepareStatement(Queries.slettVedtakISak).apply {
-                setLong(1, sakId)
-                executeUpdate()
-            }
+        utbetalingsperioder.forEach {
+            repositoryProxy.opprett(
+                query = "INSERT INTO utbetalingsperiode(vedtakid, datofom, datotom, type, beloep) " +
+                    "VALUES (:vedtakid, :datofom, :datotom, :type, :beloep)",
+                params = mapOf(
+                    "vedtakid" to vedtakId,
+                    "datofom" to it.periode.fom.atDay(1).let(Date::valueOf),
+                    "datotom" to it.periode.tom?.atEndOfMonth()?.let(Date::valueOf),
+                    "type" to it.type.name,
+                    "beloep" to it.beloep
+                ),
+                loggtekst = "Attesterer vedtak"
+            )
```

Comment:
For batch-lagring, kunne man noe ala dette?:
```
val params = utbetalingsperioder.map {
    mapOf<String, Serializable?>(
        "vedtakid" to vedtakId,
        "datofom" to it.periode.fom.atDay(1).let(Date::valueOf),
        "datotom" to it.periode.tom?.atEndOfMonth()?.let(Date::valueOf),
        "type" to it.type.name,
        "beloep" to it.beloep
    )
}
repositoryProxy.oppdaterFlere(
    query = "INSERT INTO utbetalingsperiode(vedtakid, datofom, datotom, type, beloep) VALUES (:vedtakid, :datofom, :datotom, :type, :beloep)",
    params = params,
    loggtekst = "Attesterer vedtak"
)
```

```
internal fun oppdaterFlere(query: String, params: List<Map<String, Serializable?>>, loggtekst: String) =
        using(sessionOf(datasource)) { session ->
            logger.info(loggtekst)
            session.batchPreparedNamedStatement(query, params)
        }
```

Reply:
Høres lurt ut

Reply:
Ooh 💡 Skal gi det eit forsøk!